converter/service/converter.py

```diff
@@ -0,0 +1,175 @@
+header_accept = 'Accept'
+header_content_type = 'Content-Type'
+param_charset = 'charset'
+default_charset = 'utf-8'
+
+json_key_packages = 'packages'
+param_url = 'url'
+url_path = '/transform'
+
+
+def run_server(server_class=HTTPServer):
+    """Runs a builtin python server using the given server_class.
+
+    :param server_class: server
+    :type server_class: HTTPServer
+    :return: None
+    """
+    server_address = (HOST_NAME, PORT_NUMBER)
+    httpd = server_class(server_address, Handler)
+    logger.warning('Server Starts - %s:%s', HOST_NAME, PORT_NUMBER)
+    try:
+        httpd.serve_forever()
+    except KeyboardInterrupt:
+        httpd.server_close()
+        logger.warning('Server Stops - %s:%s', HOST_NAME, PORT_NUMBER)
+
+
+class Handler(BaseHTTPRequestHandler):
+    def do_GET(s):
+        """
+        Respond to the GET request. The expected format of this request is:
+                http://<host>:<port>/transform?url=<url> along with `User-Agent`
+                and `Accept` headers
+        """
+        if not urlparse(s.path).path == url_path:
+            s.send_error(HTTPStatus.BAD_REQUEST, ErrorResponse.INVALID_PATH)
+            return
+
+        if not (header_user_agent in s.headers and header_accept in s.headers):
+            s.send_error(HTTPStatus.BAD_REQUEST, ErrorResponse.INVALID_HEADERS)
+            return
+
+        query = dict(parse_qsl(urlparse(s.path).query))
+        if param_url not in query:
+            s.send_error(HTTPStatus.BAD_REQUEST, ErrorResponse.INVALID_PARAM)
+            return
+
+        user_agent = s.headers.get(header_user_agent)
+        accept = s.headers.get(header_accept)
+        decoded_url = query.get(param_url)
+
+        try:
+            json_response = handle(decoded_url, user_agent, accept)
+            s.send_response(HTTPStatus.OK)
+            content_header = gen_universe.format_universe_repo_content_type(
+                _get_repo_version(accept))
+            s.send_header(header_content_type, content_header)
+            s.end_headers()
+            s.wfile.write(json_response.encode())
+        except Exception as e:
+            s.send_error(HTTPStatus.BAD_REQUEST, str(e))
+
+
+def handle(decoded_url, user_agent, accept):
+    """Results in either an error or the requested json data
+
+    :param decoded_url: The url to be fetched from
+    :type decoded_url: str
+    :param user_agent: User-Agent header value
+    :type user_agent: str
+    :param accept: Accept header value
+    :return Requested json data
+    :rtype str (a valid json object)
+    """
+    repo_version = _get_repo_version(accept)
+    dcos_version = _get_dcos_version(user_agent)
+    logger.debug('Url %s\nAgent %s\nAccept %s\nVersion %s\nDC/OS %s',
+                 decoded_url, user_agent, accept, repo_version, dcos_version)
+
+    req = Request(decoded_url)
+    req.add_header(header_user_agent, user_agent)
+    req.add_header(header_accept, accept)
+    res = urlopen(req)
```

Comment:
This should specify a `timeout` value, some googling implies that the default could be no timeout